### PR TITLE
[dag] commit signer

### PR DESCRIPTION
**Pull request by ibalajiarun**
### Description

This PR introduces a commit signer struct that implements CommitSignerProvider to sign buffer manager commit proofs for DAG produced blocks.

### CODE REVIEW

**Comment by ibalajiarun** [comment]
Current dependencies on/for this PR:
* main
  * **PR #9830** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9830" target="_blank">Graphite</a> 
    * **PR #9831** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9831" target="_blank">Graphite</a> 
      * **PR #9724** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9724" target="_blank">Graphite</a> 
        * **PR #9968** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9968" target="_blank">Graphite</a> 
          * **PR #9967** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9967" target="_blank">Graphite</a> 
            * **PR #10006** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10006" target="_blank">Graphite</a> 
              * **PR #9994** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9994" target="_blank">Graphite</a> 
                * **PR #10040** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10040" target="_blank">Graphite</a>  👈
                  * **PR #10041** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10041" target="_blank">Graphite</a> 
                    * **PR #10042** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10042" target="_blank">Graphite</a> 
                * **PR #10007** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10007" target="_blank">Graphite</a> 
        * **PR #9832** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9832" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10040?utm_source=stack-comment).

File: consensus/src/epoch_manager.rs [hunk 1]
```diff
@@ -1198,3 +1201,18 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
         }
     }
 }
+
+fn new_signer_from_storage(
+    author: Author,
+    backend: &SecureBackend,
+) -> aptos_types::validator_signer::ValidatorSigner {
```
Comment (zekun000) [inline consensus/src/epoch_manager.rs:1208]: we probably should get one Arc\<signer\> to share instead of parsing the backend multiple times

**Review by zekun000 — APPROVED** [review]

File: consensus/src/dag/commit_signer.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+impl CommitSignerProvider for DagCommitSigner {
+    fn sign_commit_vote(
+        &self,
+        _ledger_info: aptos_types::ledger_info::LedgerInfoWithSignatures,
+        new_ledger_info: aptos_types::ledger_info::LedgerInfo,
+    ) -> Result<bls12381::Signature, aptos_safety_rules::Error> {
+        let signature = self
+            .signer
+            .sign(&new_ledger_info)
+            .map_err(|err| aptos_safety_rules::Error::SerializationError(err.to_string()))?;
```
Comment (sasha8) [inline consensus/src/dag/commit_signer.rs:24]: We need to delete aptos_safety_rules. Can we move the error somewhere else?

File: consensus/src/dag/commit_signer.rs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+use crate::experimental::signing_phase::CommitSignerProvider;
+use aptos_crypto::bls12381;
+use aptos_types::validator_signer::ValidatorSigner;
+
+pub struct DagCommitSigner {
+    signer: ValidatorSigner,
+}
+
+impl DagCommitSigner {
+    pub fn new(signer: ValidatorSigner) -> Self {
```
Comment (sasha8) [inline consensus/src/dag/commit_signer.rs:10]: Why do we need to wrap ValidatorSigner?
So we can create a separate crate in the future?

**Review by sasha8 — APPROVED** [review]

File: consensus/src/dag/commit_signer.rs (hunk 2 above)
Comment (ibalajiarun) [reply consensus/src/dag/commit_signer.rs:24]: I think so. If and when we actually delete it, we can move?

Comment (sasha8) [reply consensus/src/dag/commit_signer.rs:24]: Sure.

**Pull request merged**